### ISSUE TITLE
Implement parent module macro

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#13434]
The purpose of this issue is to implement a macro for the parent module section of a module. For example, the following macro
```xml
<subsection name="Parent Module" id="Parent_Module">
  <macro name="parent-module">
    <param name="moduleName" value="AnnotationLocation"/>
  </macro>
</subsection>
```

should translate to
```xml
<subsection name="Parent Module" id="Parent_Module">
  <p> <a href="../../config.html#TreeWalker">TreeWalker</a> </p>
</subsection>
```

**Comment by nrmancuso** [comment]
This seems like a lot of boilerplate, is there some way to reduce this to only the macro section?

**Comment by romani** [comment]
Yes, unfortunate case, generated content is less then macros is size :).
But may be in future it will be different.

I hope one day we will find way to not define module name in macroses and have global definition somewhere  ones in whole file. May be we can use value of file name.

But I am ok to think on this at the end, when all is done and we see all edge cases.

**Comment by stoyanK7** [comment]
@nrmancuso 
> is there some way to reduce this to only the macro section?

What do you mean exactly?

> This seems like a lot of boilerplate

It is, I agree.

**Comment by nrmancuso** [comment]
>>is there some way to reduce this to only the macro section?

>What do you mean exactly?

Perhaps have the macro create the subsection too?

**Comment by romani** [comment]
If other macroses not making subsection, better to not do this and be consistent.
In current state of project wordy macros is ok, we should make it work first and then polish and optimize

**Comment by nrmancuso** [comment]
> If other macroses not making subsection, better to not do this and be consistent. In current state of project wordy macros is ok, we should make it work first and then polish and optimize

Yes, this is a good point, we can refine later

**Comment by nrmancuso** [comment]
Closed via #13435 13435
